framework/src/org/apache/cordova/ICordovaCookieManager.java

```diff
@@ -0,0 +1,45 @@
+/*
+       Licensed to the Apache Software Foundation (ASF) under one
+       or more contributor license agreements.  See the NOTICE file
+       distributed with this work for additional information
+       regarding copyright ownership.  The ASF licenses this file
+       to you under the Apache License, Version 2.0 (the
+       "License"); you may not use this file except in compliance
+       with the License.  You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+       Unless required by applicable law or agreed to in writing,
+       software distributed under the License is distributed on an
+       "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+       KIND, either express or implied.  See the License for the
+       specific language governing permissions and limitations
+       under the License.
+*/
+
+package org.apache.cordova;
+
+public interface ICordovaCookieManager {
```

Comment:
My plan was to make it mostly API-compatible with existing CookieManagers so that it would be easier to migrate existing plugins rather than trying to figure out a different API.

One other issue is that the XWalkCookieManager doesn't appear to have setThirdPartyCookiesEnabled(): https://github.com/crosswalk-project/crosswalk/blob/master/runtime/android/core_internal/src/org/xwalk/core/internal/XWalkCookieManager.java

If those aren't concerns (I guess XWalkCordovaCookieManager could just ignore setThirdPartyCookiesEnabled?), then I'll shrink down the interface.